ttnn/cpp/ttnn/operations/data_movement/bcast/device/bcast_device_operation.hpp

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/operation.hpp"
+
+namespace ttnn {
+
+enum class BcastOpMath { ADD, SUB, MUL };
+enum class BcastOpDim { H, W, HW };
+
+} // namespace ttnn
+
+
+namespace ttnn::operations::data_movement {
+
+struct BcastOperation {
+    static Tensor invoke(
+        uint8_t queue_id,
+        const Tensor &input_tensor_a,
+        const Tensor &input_tensor_b,
+        ttnn::BcastOpMath bcast_op,
+        ttnn::BcastOpDim bcast_dim,
+        const std::optional<MemoryConfig> &memory_config = std::nullopt,
+        const std::optional<Tensor> optional_output_tensor = std::nullopt);
+};
+
+
+namespace bcast_op_utils {
+  std::map<std::string, std::string> get_defines(ttnn::BcastOpDim bcast_dim, ttnn::BcastOpMath bcast_math);
+}
+
+
+
+enum class BcastOpParallelizationStrategy {
+    MULTI_CORE_H_SHARDED,
+    MULTI_CORE_H_SHARDED_OPTIMISED,
+    MULTI_CORE_H,
+    MULTI_CORE_W,
+    MULTI_CORE_HW,
+    SINGLE_CORE
+};
+
+
+operation::ProgramWithCallbacks bcast_multi_core_h(
```

Comment:
Ideally all these program things are also separated from this header. Ideally each program is separate pair of hpp / cpp. 
Headers of needed programs than included in device operation cpp file remaining a detail of implementation

Reply:
yeah this absolutely should be done... splitting it now